genum-derive/src/lib.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright 2017-2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+extern crate proc_macro;
+
+use anyhow::{bail, Result};
+use heck::{KebabCase, SnakeCase};
+use itertools::Itertools;
+use proc_macro2::TokenStream;
+use proc_macro_error::{abort_call_site, proc_macro_error};
+use quote::{format_ident, quote, quote_spanned};
+use syn::{
+    parse_macro_input, punctuated::Punctuated, spanned::Spanned, token::Comma, Attribute, Data,
+    DeriveInput, Ident, Lit, Meta, MetaList, NestedMeta, Variant,
+};
+
+#[proc_macro_derive(GEnum, attributes(genum))]
+#[proc_macro_error]
+pub fn genum_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
```

Comment:
Please put this macro into a separate module instead of directly in `lib.rs`

Reply:
Aren't proc macros supposed to be put directly in `lib.rs`?

Reply:
Not sure, but even if so you could put at least the implementation (what you call in here) into a separate module :)

I'm mostly worried about having `lib.rs` becoming huge once we add more macros